Add user field to artifact events and download and delete events (#172)
cdevents/spec: continuous-integration.md

-->
# Continuous Integration Events

> Continuous Integration (CI) events include the subject and predicates related to CI activities such as building software, producing [artifacts](./artifacts.md) and [running tests](./testing-events.md).

Author: Thanks, fixed now

| type | `String` | See [type](spec.md#type-subject) | `artifact` | |
| change | `object`        | The change (tag, commit, revision) of the repository which was used to build the artifact" | `{"id": "527d4a1aca5e8d0df24813df5ad65d049fc8d312", "source": "my-git.example/an-org/a-repo"}`, `{"id": "feature1234", "source": "my-git.example/an-org/a-repo"}` | ✅ |

### [`artifact signed`](examples/artifact_signed.json)

The event represents an artifact that has been signed. The signature is included in the events itself.
An artifact may be signed after it has been packaged or sometimes after it has published, depending on the tooling being used and the type of artifact. The `artifact signed` event is typically produced by the CI or build system.

> - Event Type: __`dev.cdevents.artifact.signed.0.1.0`__

Reviewer: ```suggestion
- Event Type: __`dev.cdevents.artifact.signed.0.1.0-draft`__
```

Author: This is not a new event and it has not been modified in the PR, so the version should not change.

An artifact may be signed after it has been packaged or sometimes after it has published, depending on the tooling being used and the type of artifact. The `artifact signed` event is typically produced by the CI or build system.

- Event Type: __`dev.cdevents.artifact.signed.0.1.0`__
- Predicate: signed
- Subject: [`artifact`](#artifact)

| Field | Type | Description | Examples | Required |
|-------|------|-------------|----------|----------------------------|
| id    | `Purl` | See [id](spec.md#id-subject) | `pkg:oci/myapp@sha256%3A0b31b1c02ff458ad9b7b81cbdf8f028bd54699fa151f221d1e8de6817db93427?repository_url=mycr.io/myapp`, `pkg:golang/mygit.com/myorg/myapp@234fd47e07d1004f0aed9c` | ✅ |
| source | `URI-Reference` | See [source](spec.md#source-subject) | | |
| type | `String` | See [type](spec.md#type-subject) | `artifact` | |
| signature | `string`     | The signature of the artifact | `MEYCIQCBT8U5ypDXWCjlNKfzTV4KH516/SK13NZSh8znnSMNkQIhAJ3XiQlc9PM1KyjITcZXHotdMB+J3NGua5T/yshmiPmp` | ✅ |

### [`artifact published`](examples/artifact_published.json)

The event represents an artifact that has been published and it can be advertised for others to use.

> The `artifact signed` event is typically produced by the artifact registry, but it may also be produced by the build system.

Reviewer: ```suggestion
The `artifact published` event is typically produced by the artifact registry, but it may also be produced by the build system.
```

Author: Fixed

| id    | `Purl` | See [id](spec.md#id-subject) | `pkg:oci/myapp@sha256%3A0b31b1c02ff458ad9b7b81cbdf8f028bd54699fa151f221d1e8de6817db93427?repository_url=mycr.io/myapp`, `pkg:golang/mygit.com/myorg/myapp@234fd47e07d1004f0aed9c` | ✅ |
| source | `URI-Reference` | See [source](spec.md#source-subject) | | |
| type | `String` | See [type](spec.md#type-subject) | `artifact` | |
| user | `String` | The user who published to the artifact registry. [^user] | `mybot-myapp` | |

### [`artifact downloaded`](examples/artifact_downloaded.json)

> The event represents an artifact that has been published and it can be advertised for others to use.

Reviewer: ```suggestion
The event represents an artifact that has been downloaded from the registry.
```

Author: Done

> The `artifact signed` event is typically produced by the artifact registry, but it may also be produced by the build system.

Author: Done

| id    | `Purl` | See [id](spec.md#id-subject) | `pkg:oci/myapp@sha256%3A0b31b1c02ff458ad9b7b81cbdf8f028bd54699fa151f221d1e8de6817db93427?repository_url=mycr.io/myapp`, `pkg:golang/mygit.com/myorg/myapp@234fd47e07d1004f0aed9c` | ✅ |
| source | `URI-Reference` | See [source](spec.md#source-subject) | | |
| type | `String` | See [type](spec.md#type-subject) | `artifact` | |
| user | `String` | The user who published to the artifact registry. [^user] | `mybot-myapp` | |

### [`artifact downloaded`](examples/artifact_downloaded.json)

The event represents an artifact that has been downloaded from the registry.

> The `artifact downloaded` event is typically produced by the artifact registry, but it may also be produced by the build system.

Reviewer: It's probably more common that a test system or a CI system in general downloads an artifact, than a build system (except for libraries and things to integrate in builds of course)
```suggestion
The `artifact downloaded` event is preferably produced by the artifact registry.
```

Author: Nice suggestion, thank you.

- Event Type: __`dev.cdevents.artifact.downloaded.0.1.0-draft`__
- Predicate: downloaded
- Subject: [`artifact`](#artifact)

| Field | Type | Description | Examples | Required |
|-------|------|-------------|----------|----------------------------|
| id    | `Purl` | See [id](spec.md#id-subject) | `pkg:oci/myapp@sha256%3A0b31b1c02ff458ad9b7b81cbdf8f028bd54699fa151f221d1e8de6817db93427?repository_url=mycr.io/myapp`, `pkg:golang/mygit.com/myorg/myapp@234fd47e07d1004f0aed9c` | ✅ |
| source | `URI-Reference` | See [source](spec.md#source-subject) | | |
| type | `String` | See [type](spec.md#type-subject) | `artifact` | |
| user | `String` | The user who downloaded from the artifact registry. [^user] | `mybot-myapp` | |

### [`artifact deleted`](examples/artifact_deleted.json)

The event represents an artifact that has been deleted from an artifact registry.

> The `artifact deleted` event is typically produced by the artifact registry, but it may also be produced by the build system.

Reviewer: ```suggestion
The `artifact deleted` event is preferably produced by the artifact registry.
```
